docs/vector_search.rst

```diff
@@ -0,0 +1,278 @@
+
+An example where this particular use case may make sense is a scenario where one would want to find burger photos for a particular business. The vector search query can find its top burger photo across all businesses. The text search can apply a filter based on business ID and caption field of the document. If there are photos from the same business in the vector search, their score can be boosted using the boost parameter so that when combined with the regular text search results, they get higher score. In this case even if no photos are found from the vector search, the text search can at least show some photos whose caption matches the keyword "burger".
+
+Vector Search + Top Level Filter
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Traverses through HSNW graph + Does an OR with the results from the top level query statement. Can be used for boosting on top of existing lucene score.
+The results are less accurate. Example::
+
+    {
+      "indexName": "vector_test",
+      "startHit": 0,
+      "topHits": 10,
+      "timeoutSec": 0,
+      "retrieveFields": [
+        "photo_id",
+        "business_id",
+        "caption"
+      ],
+      "query": {
+        "booleanQuery": {
+          "clauses": [
+            {
+              "occur": "MUST",
+              "query": {
+                "termQuery": {
+                  "field": "business_id",
+                  "intValue": 12581436
+                }
+              }
+            }
+          ]
+        }
+      },
+      "knn": [
+        {
+          "field": "photo_embeddings",
+          "k": 100,
+          "num_candidates": 1000,
+          "query_vector": [
+            0.188423157,
+            -0.0844727457,
+        ....
+       ]
+      }
+     ]
+    }
+
+
+Optimizing Search Queries
+=========================
+
+The vector hits value represents the number of documents traversed during the vector search. It is the number of vector comparisons, which is the major factor in query performance. It plays the most important role in terms of search latencies and accuracy.
+Any change that reduces the vector hits number, will decrease the latencies in expense of reducing accuracy.
+
+A summary of trade-offs for each config:
+
+* Improve Search Latency
+
+  * Lower `num_candidates`
+
+    * Lower vector hits (latency)
+    * Lower accuracy
+
+  * Lower indexing parameter values
```

Comment:
Might be good to mention what parameters these are, in this section somewhere

Reply:
Added